### ISSUE TITLE
Measuring is affected by current transformation; should it be?

### ISSUE THREAD

**Issue by jhauberg** [jhauberg/termlike#30]
For example, in the following snippet, we expect a text with a solid background- and that is what we get:

```c
char const * const name = "TARGETING DRONE MKII";

struct term_dimens name_dims;

term_measure(name, &name_dims);

term_fill(positioned(10, 10), name_dims, TERM_COLOR_WHITE);
term_print(name, positioned(10, 10), TERM_COLOR_BLACK);
```

And that is what we get:

![screenshot 2018-11-09 at 09 16 51](https://user-images.githubusercontent.com/1360637/48251019-907c5380-e400-11e8-9968-1105f41c1a86.png)

However, what if we want to scale the text down? We might do the following:

```c
char const * const name = "TARGETING DRONE MKII";

term_set_transform(scaled(0.8f));

struct term_dimens name_dims;

term_measure(name, &name_dims);

term_fill(positioned(10, 10), name_dims, TERM_COLOR_WHITE);
term_print(name, positioned(10, 10), TERM_COLOR_BLACK);
```

But the output is probably not what was expected:

![screenshot 2018-11-09 at 09 21 01](https://user-images.githubusercontent.com/1360637/48251129-e4873800-e400-11e8-8aea-124ce5b39503.png)

*Notice how the "MKII" is seemingly being cut off?*

What happens here, is that both the fill and the print is being scaled, but since the dimensions given to `term_fill` is already taking the scaling into account, the resulting rect is essentially being scaled down twice.

So it is pretty easy to fix, but it might not be immediately obvious why it didn't work in the first place. Anyway, the easy fix is to measure the text before applying a transformation:

```c
char const * const name = "TARGETING DRONE MKII";

struct term_dimens name_dims;

term_measure(name, &name_dims);

term_set_transform(scaled(0.8f));

term_fill(positioned(10, 10), name_dims, TERM_COLOR_WHITE);
term_print(name, positioned(10, 10), TERM_COLOR_BLACK);
```

![screenshot 2018-11-09 at 09 27 59](https://user-images.githubusercontent.com/1360637/48251396-c5d57100-e401-11e8-9625-d2f008b308e3.png)

So this begs the question: should `term_measure` be affected by the current transformation?

Rotation is already obviously a non-factor, but what about scaling? I think there are reasons internally to require the final resulting size to be known, but we can probably find a way to avoid that, or just keep it internal-only. I'm not sure of the consequences for typical usage if this was to be changed.

Basically, does it work as expected?

**Comment by jhauberg** [comment]
*Alternatively*, what if `term_fill` was not affected by transformation?

**Comment by jhauberg** [comment]
Closing for now.
